pandas/tests/resample/test_resampler_grouper.py

```diff
@@ -470,3 +470,30 @@ def test_resample_groupby_agg_object_dtype_all_nan(consolidate):
         index=idx,
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_resample_with_list_of_keys():
+    # GH 47362
+    df = DataFrame(
+        data={
+            "date": date_range(start="2016-01-01", periods=8),
+            "group": [0, 0, 0, 0, 1, 1, 1, 1],
+            "val": [1, 7, 5, 2, 3, 10, 5, 1],
+        }
+    )
+    result = df.groupby("group").resample("2D", on="date")[["val"]].mean()
```

Comment:
Do we have tests for the `[["val", "date"]]`` case too?

Reply:
No. I will do this.

Reply:
Here is something ambiguous in the original issue. The author said we should get the same result when pass `["val", "date"]` and `["val"]`. But I think this is not appropriate since we will get mean of the datetime if we only pass ["date"]. Thus actually `["val", "date"]` should also return 2 columns instead of one. This behavior is also consistent with `groupby`, something like `df.groupby("key")[["key", "val"]].mean()`, which also returns 2 aggregation columns.

Reply:
I agree that `["val", "date"]` should return 2 columns